### PR TITLE
refactor(console): improve m2m app guide content

### PR DESCRIPTION
**Pull request by charIeszhao**
## Summary

Improve M2M app guide content. Previously, there were issues with the endpoint display.

image

- The endpoint should display custom domain first, instead of always showing the default tenant ID
- Should not have double slashes in endpoint URL
- The basic authorization credential should use the base64 encoded value of the real-time app id and app secret combination.

## Testing

Locally tested

## Checklist

~~- [ ] `.changeset`~~
~~- [ ] unit tests~~
~~- [ ] integration tests~~
~~- [ ] docs~~

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: packages/console/src/assets/docs/guides/m2m-general/README.mdx [hunk 1]
```diff
@@ -54,12 +57,12 @@ In the API Resource tab, find the API identifier that the app needs to access. I
...
       Basic authentication
     </a>
-    , where username is the App ID, and password is the App Secret.
+    , where username is the App ID, and password is the App Secret. The credential is the username and password combined with a colon, encoded in base64.
   </li>
   <li>Carry the body data</li>
 </ul>
 
-```jsonc
+```json
```
Comment (gao-sun) [inline packages/console/src/assets/docs/guides/m2m-general/README.mdx:65]: should be jsonc (stands for "josn comment")?

File: packages/console/src/assets/docs/guides/m2m-general/README.mdx [hunk 2]
```diff
@@ -54,12 +57,12 @@ In the API Resource tab, find the API identifier that the app needs to access. I
     <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization#basic_authentication">
       Basic authentication
     </a>
-    , where username is the App ID, and password is the App Secret.
+    , where username is the App ID, and password is the App Secret. The credential is the username and password combined with a colon, encoded in base64.
```
Comment (gao-sun) [inline packages/console/src/assets/docs/guides/m2m-general/README.mdx:60]: it'll be great if we can have an example here

**Review by gao-sun — APPROVED** [review]

File: packages/console/src/assets/docs/guides/m2m-general/README.mdx (hunk 2 above)
Comment (charIeszhao) [reply packages/console/src/assets/docs/guides/m2m-general/README.mdx:60]: Alright, I guess I can write the code here as an example. And we do calculate the base64 value for users in the code sample below

**Review by wangsijie — APPROVED** [review]

File: packages/console/src/assets/docs/guides/m2m-general/README.mdx (hunk 2 above)
Comment (charIeszhao) [reply packages/console/src/assets/docs/guides/m2m-general/README.mdx:60]: No, I changed my mind. It reads too wordy.

File: packages/console/src/assets/docs/guides/m2m-general/README.mdx (hunk 1 above)
Comment (charIeszhao) [reply packages/console/src/assets/docs/guides/m2m-general/README.mdx:65]: Our code editor doesn't seem to support jsonc, and it was totally plain text
image
